Tests/AsyncHTTPClientTests/HTTPClientTests.swift

```diff
@@ -709,6 +710,69 @@ class HTTPClientTests: XCTestCase {
         }
     }
 
+    func testProxySOCKS() throws {
+        let socksBin = try MockSOCKSServer(expectedURL: "/socks/test", expectedResponse: "it works!")
+        let localClient = HTTPClient(eventLoopGroupProvider: .shared(self.clientGroup),
+                                     configuration: .init(proxy: .socksServer(host: "127.0.0.1")))
+
+        defer {
+            XCTAssertNoThrow(try localClient.syncShutdown())
+            XCTAssertNoThrow(try socksBin.shutdown())
+        }
+
```

Comment:
Can we move those tests into an extra test file `HTTPClient+SOCKSTests` or something along the lines? `HTTPClientTests` is slowly approaching 3k loc. And I think more separation will make dealing with test easier going forward.